name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/Bank11PDFExtractor.java

```diff
@@ -0,0 +1,107 @@
+                                        .section("year").match("(.*)(Kontoauszug Nr\\.)(\\s*)(\\d+)\\/(?<year>\\d{4})")
+                                        .assign((ctx, v) -> ctx.put("year", v.get("year")))
+
+                                        .section("currency").match("(?<currency>[\\w]{3})(-Konto Kontonummer)(.*)")
+                                        .assign((ctx, v) -> ctx.put(CONTEXT_KEY_CURRENCY,
+                                                        asCurrencyCode(v.get("currency")))));
+        this.addDocumentTyp(type);
+
+        Block depositBlock = new Block(DEPOSIT);
+        depositBlock.set(depositTransaction(type, DEPOSIT));
+        type.addBlock(depositBlock);
+
+        Block removalBlock = new Block(REMOVAL);
+        removalBlock.set(removalTransaction(type, REMOVAL));
+        type.addBlock(removalBlock);
+
+        Block interestBlock = new Block(INTEREST);
+        interestBlock.set(interestTransaction(type, INTEREST));
+        type.addBlock(interestBlock);
+    }
+
+
+    private Transaction<AccountTransaction> depositTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DEPOSIT);
+            return entry;
+        }).section("date", "amount").match(regex).assign(assignmentsProvider(type)).wrap(TransactionItem::new);
+    }
+
+    private Transaction<AccountTransaction> interestTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.INTEREST);
+            return entry;
+        }).section("date", "amount").match(regex).assign(assignmentsProvider(type)).wrap(TransactionItem::new);
+    }
+
+
+    private Transaction<AccountTransaction> removalTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.REMOVAL);
+            return entry;
+        }).section("date", "amount").match(regex).assign(assignmentsProvider(type)).wrap(TransactionItem::new);
+    }
+
+
+    private BiConsumer<AccountTransaction, ParsedData> assignmentsProvider(DocumentType type)
+    {
+        return (transaction, matcherMap) -> {
+            Map<String, String> context = type.getCurrentContext();
+
+            String date = matcherMap.get("date");
+
+            date += context.get(CONTEXT_KEY_YEAR);
+
+            transaction.setDateTime(asDate(date));
+            transaction.setAmount(asAmount(matcherMap.get("amount")));
+            transaction.setCurrencyCode(asCurrencyCode(context.get(CONTEXT_KEY_CURRENCY)));
+        };
+    }
```

Comment:
Hello @ZfT2 
1.)
I think you keep making the same mistake...

2.)
You dont need the BiConsumer.... again....

3.)
You dont need `Map<String, String> context = type.getCurrentContext();`

add follow like -> [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DkbPDFExtractor.java#L723:L738) <-

`.documentContext("currency", "year") //`

then you can simply use `v.get("currency")` or `v.get("year")` to retrieve the string, as it is already parsed in the document.